source/_drafts/Introducing-Thumbcoil.md

```diff
@@ -0,0 +1,77 @@
+
+{% pullquote left %}
+MP4 files are composed of **boxes** - hierarchical logical units that, conveniently, all start with a 32-bit length and a 32-bit box-type. Boxes will often contain other sub-boxes.
+{% endpullquote %}
+
+The answer to that problem was to build an "mp4-inspector" - a tool that would parse MP4 and display a sort of JSON-like dump of any relevant boxes and their contents. By generating a dump of the output from Mux.js and comparing it to a known-good fragment generated with FFmpeg, we could see where our transmuxer's output differed.
+
+The "mp4-inspector" was built as a web page so that we can have a graphical color-coded diff of the two segments. Over time the page gained an video element and we started appending the results of transmuxing segments directly into the video element's media-source to aid in instant feedback and validation of changes to Mux.js.
+
+## A Brave New World
+{% pullquote right %}
+A **media container** such as MP4 encapsulates the video and audio stream. It has metadata describing the streams, timing information for each frame, and the stream data itself.
+{% endpullquote %}
+
+As development continued, we would sometimes encounter streams that would fail in new and interesting ways. Some of these failures were, admittedly, due to bugs in Mux.js but as development continued and Mux.js itself became more robust they were increasingly caused by problems with the streams or issues with a particular implementation of the media-source-extensions specification.
+
+It eventually dawned on us that we really needed to learn more about what was happening inside of those videos. We needed to see not just what was happening at the media container level but we had to go deeper - we needed to peek into the video data itself. For that purpose we created Thumbcoil.
+
+{% pullquote left %}
+Inside of a container, video and audio are contained in data called **bitstreams**. Bitstreams are the data produced by encoders to represent the audio signals or video frames. Some common bitstreams are AAC for audio and H.264 for video.
+{% endpullquote %}
+
+## Thumbcoil
+
+Thumbcoil is a suite of tools designed to give you a peek into the internals of H.264 video bitstreams contained inside either an MPEG4 or MPEG2-TS container files. Using the tools in Thumbcoil you can get a detailed view of the internal structure of the two supported media container formats.
+
+In addition, the tools have the ability to show you the information contained within the most important NAL-units that make up the H.264 bitstream. Ever wonder what kind of secret information the video encoder has squirreled away for decoders to use? Now, with Thumbcoil, you can finally see for yourself!
+
+## Motivation
+{% pullquote right %}
+An H.264-encoded bitstream is composed of what are called **NAL units** or network-abstraction-layer. NALs are a simple packet format designed to use bits as efficiently as possible.
```

Comment:
No need for hyphenation.
Also, "NAL units" would end up being "network abstraction layer units".